napari/plugins/_builtins.py

```diff
@@ -27,3 +36,200 @@ def napari_get_reader(path: Union[str, List[str]]) -> ReaderFunction:
         function that returns layer_data to be handed to viewer._add_layer_data
     """
     return image_reader_to_layerdata_reader(magic_imread)
+
+
+@napari_hook_implementation(trylast=True)
+def napari_write_image(path: str, data: Any, meta: dict) -> Optional[str]:
+    """Our internal fallback image writer at the end of the plugin chain.
+
+    Parameters
+    ----------
+    path : str
+        Path to file, directory, or resource (like a URL).
+    data : array or list of array
+        Image data. Can be N dimensional. If meta['rgb'] is ``True`` then the
+        data should be interpreted as RGB or RGBA. If ``meta['is_pyramid']`` is
+        ``True``, then the data should be interpreted as an image pyramid.
+    meta : dict
+        Image metadata.
+
+    Returns
```

Comment:
I know changes are still in progress but just flagging so as not to forget.
Doc string here needs update I think
```
    Returns
    -------
    path : str or None
        If data is successfully written, return the ``path`` that was written.
        Otherwise, if nothing was done, return ``None``.
    """
```